src/com/owncloud/android/lib/resources/shares/OCShare.java

```diff
@@ -306,6 +316,7 @@ public void readFromParcel(Parcel source) {
         mUserId = source.readLong();
         mRemoteId = source.readLong();
         mShareLink = source.readString();
+        mIsPasswordProtected = source.readInt() == 0;
```

Comment:
We are writing a 1 if mIsPasswortProtected is true in line 346, but compare with 0?
So if mIsPasswortProteced = true, then
on write: dest.writeInt(1).
on read: source.readInt() == 0 --> 1 == 0 --> false --> mIsPasswordProtected = false